### PR TITLE
InstanceMutator Machine API for SetModificationStatus

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following exposes SetModificationStatus on the instance mutator
machine api.

## QA steps

Note to actually see the changes, edit the `worker/instancemutater/mutater.go`
file and add a call to the `machine.SetModificationStatus`

Then perform the following:

```
juju bootstrap lxd test
juju deploy ./testcharms/charm-repo/quantal/lxd-profile
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/TPASyp5k/1198-instancemutater-setmodificationstatus-for-the-api

File: api/instancemutater/machine.go [hunk 1]
```diff
@@ -40,6 +41,11 @@ type MutaterMachine interface {
 	// WatchUnits returns a watcher.StringsWatcher for watching units of a given
 	// machine.
 	WatchUnits() (watcher.StringsWatcher, error)
+
+	// SetModificationStatus sets the provider specific modification status
+	// for a machine. Allowing the propergation of status messages to the
```
Comment (hmlanigan) [inline api/instancemutater/machine.go]: s/propergation/propagation/

File: state/machine.go [hunk 2]
```diff
@@ -1234,8 +1234,9 @@ func (m *Machine) ModificationStatus() (status.StatusInfo, error) {
 	return machineStatus, nil
 }
 
-// SetModificationStatus sets the provider specific modification status for a
-// machine.
+// SetModificationStatus sets the provider specific modification status
+// for a machine. Allowing the propergation of status messages to the
```
Comment (hmlanigan) [inline state/machine.go]: s/propergation/propagation/

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**